File: 1871-jump-game-vii/1871-jump-game-vii.cpp
class Solution {
public:
  bool canReach(string s, int minJump, int maxJump) {
    int l = s.length();
    int c = 0;
    int total[l+1];
	  memset(total, 0, sizeof total);
    
    if (s.back() == '1') {
      return false;
    }
    
    for (int i = 0; i < l; ++i) {
      c += total[i];
      if (i == 0 || (c && s[i] == '0')) {
        ++total[min(i + minJump, l)];
        --total[min(i + maxJump + 1, l)];
      }
    }
    
    return c;
  }
};

File: 1871-jump-game-vii/1871-jump-game-vii.py
class Solution:
  def canReach(self, s: str, minJump: int, maxJump: int) -> bool:
    l = len(s)
    c = 0
    total = [0 for _ in range(l+1)]
    
    if s[-1] == '1':
      return False
    
    for i in range(l):
      c += total[i];
      if i == 0 or (c and s[i] == '0'):
        total[min(i + minJump, l)] += 1;
        total[min(i + maxJump + 1, l)] -= 1;
    
    return c;
    